instrumentation-annotations-support/src/main/java/io/opentelemetry/instrumentation/api/annotation/support/async/AsyncOperationEndHandler.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.annotation.support.async;
+
+import io.opentelemetry.context.Context;
+import javax.annotation.Nullable;
+
+/** Callback that is called when async computation completes. */
+public interface AsyncOperationEndHandler<REQUEST, RESPONSE> {
```

Comment:
I get the name, but I think it's too specific and long. 
What else can it handle apart from ending a span?

Reply:
I guest it needs to handle the end of the timer and record the duration to the histogram metrics recorder.

Reply:
This is effectively an Instrumentation Callback... Why not call it that?
Something like `InstrumentationCallback` with an `end()` method... 
I can imagine this also being useful to add attributes if later we add an `addAttribute()` method.

Reply:
Hi @brunobat, I go through the code. The class `Instrumentation` is coupled with the implementation of the span's operations([here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/ffeb80eb26aeba40fd4339c6c2113572b94a6668/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java#L148)) and it's widely used. It's going far from the guide's suggestion([here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/using-instrumenter-api.md#using-the-instrumenter-api)). I think to create a new set of abstract operation for metrics instrumentation and update the guide would be an option. @laurit gave a good example of it. And what's your opinion about it...

Reply:
Can we just drop the `Operation` bit from everywhere? `AsyncEndHandler` seems already clear enough, as `AsyncEndStrategy` seems good enough for me.